app/models/hostgroup.rb

```diff
@@ -195,9 +195,15 @@ def clone(name = "")
     new.puppetclasses = puppetclasses
     new.locations     = locations
     new.organizations = organizations
+    new.config_groups = self.config_groups
+
     # Clone any parameters as well
     self.group_parameters.each{|param| new.group_parameters << param.clone}
-    self.config_groups.each{|group| new.config_groups << group}
+    self.lookup_values.each do |lookup_value|
+      new_lookup_value = lookup_value.dup
+      new_lookup_value.match = "hostgroup=#{name}"
```

Comment:
Given a nested host group "Foo/Bar", its name is "Bar" but its title is "Foo/Bar".  I think the matcher needs to use the title.